### PR TITLE
SetSystemContextActivationStateAndContextAssociation manipulation

### PR DESCRIPTION
**Pull request by belagertem**
new manipulation SetSystemContextActivationStateAndContextAssociation added

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: src/t2iapi/combined/service.proto [hunk 1]
```diff
@@ -62,4 +62,14 @@ service CombinedService {
   rpc SetModeOfOperationAndSetOperatingMode (
   t2iapi.combined.SetModeOfOperationAndSetOperatingModeRequest)
       returns (BasicResponse);
+
+  /*
+  Request to set the activation state of the system context with the specified handle to the requested value and to set
```
Comment (jannast) [inline src/t2iapi/combined/service.proto:67]: I would prefer wording @ActivationState and @ContextAssociation as mostly used in this file.

File: src/t2iapi/combined/combined_requests.proto [hunk 2]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
 }
+
+/*
+  Request to set the activation state of the system context with the specified handle to the requested value and to set
+  the context association of a context state from the context descriptor with the given handle to the requested value.
+*/
+message SetSystemContextActivationStateAndContextAssociationRequest {
```
Comment (jannast) [inline src/t2iapi/combined/combined_requests.proto:78]: the specification of the interface seems to me correct. I would only adapt the wording to the biceps standard (as suggested in the other remark)

**Review by jannast — CHANGES_REQUESTED** [review]

File: src/t2iapi/combined/service.proto (hunk 1 above)
Comment (belagertem) [reply src/t2iapi/combined/service.proto]: changed

File: src/t2iapi/combined/combined_requests.proto (hunk 2 above)
Comment (belagertem) [reply src/t2iapi/combined/combined_requests.proto:78]: changed

File: src/t2iapi/combined/combined_requests.proto [hunk 3]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
...
     t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
 }
+
+/*
+Request to set the @ActivationState of the system context with the specified handle to the requested value and to set
+the @ContextAssociation of a context state from the context descriptor with the given handle to the requested value.
+*/
+message SetSystemContextActivationStateAndContextAssociationRequest {
+    string system_context_handle = 1; // handle of the system context for which the @ActivationState shall be set
+    t2iapi.activation_state.ComponentActivation system_context_activation = 2; // the @ActivationState the system context shall be set to
```
Comment (jannast) [inline src/t2iapi/combined/combined_requests.proto:80]: it is the requested ComponentActivation value

**Review by jannast — CHANGES_REQUESTED** [review]

File: src/t2iapi/combined/combined_requests.proto (hunk 3 above)
Comment (belagertem) [reply src/t2iapi/combined/combined_requests.proto]: changed the comment

**Review by jannast — APPROVED** [review]

File: src/t2iapi/combined/combined_requests.proto [hunk 4]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
 }
+
+/*
+Request to set the @ActivationState of the system context with the specified handle to the requested value and to set
+the @ContextAssociation of a context state from the context descriptor with the given handle to the requested value.
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:76]: Adjust the wording to match the other context manipulations.

```suggestion
Request to set the @ActivationState of the system context with the specified handle to the requested value and to create or adapt
the @ContextAssociation of a context state from the context descriptor with the given handle to the requested value.
```

Comment (belagertem) [reply src/t2iapi/combined/combined_requests.proto]: changed

File: src/t2iapi/combined/service.proto [hunk 5]
```diff
@@ -62,4 +62,14 @@ service CombinedService {
   rpc SetModeOfOperationAndSetOperatingMode (
   t2iapi.combined.SetModeOfOperationAndSetOperatingModeRequest)
       returns (BasicResponse);
+
+  /*
+  Request to set the @ActivationState of the system context with the specified handle to the requested value and to set
+  the @ContextAssociation of a context state from the context descriptor with the given handle to the requested value.
```
[the remark this answers is not in the document]
Comment (belagertem) [reply src/t2iapi/combined/service.proto]: changed

File: src/t2iapi/combined/combined_requests.proto [hunk 6]
```diff
@@ -70,3 +70,15 @@ message SetModeOfOperationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
 }
+
+/*
+Request to set the @ActivationState of the system context with the specified handle to the requested value and to
+create or adapt the @ContextAssociation of a context state from the context descriptor with the given handle to the
```
Comment (jannast) [inline src/t2iapi/combined/combined_requests.proto:76]: it has to be create or adapt of a context state not the ContextAssociation.
e.g.
create a new or adapt an existing context state with the requested @ContextAssociation from the context descriptor with the given handle.

**Review by jannast — CHANGES_REQUESTED** [review]

File: src/t2iapi/combined/combined_requests.proto (hunk 6 above)
Comment (belagertem) [reply src/t2iapi/combined/combined_requests.proto]: fixed the wording

**Review by jannast — DISMISSED** [review]

File: src/t2iapi/combined/combined_requests.proto [hunk 7]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
 }
+
+/*
+Request to set the @ActivationState of the system context with the specified handle to the requested value and to
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:75]:
```suggestion
Request to set the @ActivationState of the pm:SystemContextState with the specified HANDLE to the requested value and to
```

File: src/t2iapi/combined/combined_requests.proto [hunk 8]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
...
+create or adapt a context state with the requested @ContextAssociation for the context descriptor with the given handle.
+*/
+message SetSystemContextActivationStateAndContextAssociationRequest {
+    string system_context_handle = 1; // handle of the system context for which the @ActivationState shall be set
+    t2iapi.activation_state.ComponentActivation system_context_activation = 2; // the requested ComponentActivation value to set for the system context
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:80]:
```suggestion
    t2iapi.activation_state.ComponentActivation system_context_activation = 2; // the requested pm:ComponentActivation value to set for the system context
```

File: src/t2iapi/combined/combined_requests.proto [hunk 9]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4; // OperatingMode value to be set for the operation
 }
+
+/*
+Request to set the @ActivationState of the system context with the specified handle to the requested value and to
+create or adapt a context state with the requested @ContextAssociation for the context descriptor with the given handle.
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:76]:
```suggestion
create or adapt a pm:AbstractContextState with the requested @ContextAssociation for the pm:AbstractContextDescriptor with the given HANDLE.
```

File: src/t2iapi/combined/combined_requests.proto [hunk 10]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
...
+*/
+message SetSystemContextActivationStateAndContextAssociationRequest {
+    string system_context_handle = 1; // handle of the system context for which the @ActivationState shall be set
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:79]:
```suggestion
    string system_context_handle = 1; // HANDLE of the pm:SystemContextState for which the @ActivationState shall be set
```

File: src/t2iapi/combined/combined_requests.proto [hunk 11]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
...
+    t2iapi.activation_state.ComponentActivation system_context_activation = 2; // the requested ComponentActivation value to set for the system context
+    string context_descriptor_handle = 3; // handle of the context descriptor from which the @ContextAssociation of one of its context states shall be set
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:81]:
```suggestion
    string context_descriptor_handle = 3; // HANDLE of the pm:AbstractContextDescriptor for which one of its pm:AbstractContextStates shall be created or adapted with the given @ContextAssociation
```

File: src/t2iapi/combined/combined_requests.proto [hunk 12]
```diff
@@ -70,3 +70,14 @@ message SetModeOfOperationAndSetOperatingModeRequest {
...
+    t2iapi.context.ContextAssociation context_association = 4; // the @ContextAssociation to be set to a context state of the requested context descriptor
```
Comment (ldeichmann) [inline src/t2iapi/combined/combined_requests.proto:82]:
```suggestion
    t2iapi.context.ContextAssociation context_association = 4; // the @ContextAssociation to be set for the created or adapted pm:AbstractContextState of the requested pm:AbstractContextDescriptor
```

File: src/t2iapi/combined/service.proto [hunk 13]
```diff
@@ -62,4 +62,15 @@ service CombinedService {
   rpc SetModeOfOperationAndSetOperatingMode (
   t2iapi.combined.SetModeOfOperationAndSetOperatingModeRequest)
       returns (BasicResponse);
+
+  /*
+  Request to set the @ActivationState of the system context with the specified handle to the requested value and to
```
Comment (ldeichmann) [inline src/t2iapi/combined/service.proto:67]:
```suggestion
  Request to set the @ActivationState of the pm:SystemContextState with the specified HANDLE to the requested value and to
```

File: src/t2iapi/combined/service.proto [hunk 14]
```diff
@@ -62,4 +62,15 @@ service CombinedService {
...
+  create or adapt a context state with the requested @ContextAssociation for the context descriptor with the given
```
Comment (ldeichmann) [inline src/t2iapi/combined/service.proto:68]:
```suggestion
  create or adapt a pm:AbstractContextState with the requested @ContextAssociation for the pm:AbstractContextDescriptor with the given
```

File: src/t2iapi/combined/service.proto [hunk 15]
```diff
@@ -62,4 +62,15 @@ service CombinedService {
...
+  handle.
```
Comment (ldeichmann) [inline src/t2iapi/combined/service.proto:69]:
```suggestion
  HANDLE.
```

**Review by ldeichmann — COMMENTED** [review]
This is the "strict" version of using prefixes and BICEPS Terms such as HANDLE, just to see what it looks like

**Review by ldeichmann — APPROVED** [review]

**Review by jannast — APPROVED** [review]

**Pull request merged**